Improve the ./easy-setup.sh script

yodapotatofly opened the pull request:
- **not use docker group anymore (bad practice) -> run script as root instead**

The use of "docker" group is considered bad practice because, as docker engine runs as root, it gives root privileges to any user in the group. Unlike sudo, there is no password check on the docker command so this could lead to security issues.

The use of the docker group also made the script less conveniant since the user had to log off and in again for the new group to be taken into account.

- **Command line options for installing docker, compose and portainer in non interactive mode**

Following PR #343 , it can be usefull to be able to install components (docker, compose or portainer) during a non interactive run of the script. Therefore, I'm introducing `--install-docker`,  `--install-docker-compose`,  `--install-portainer` and `--install-all`
 options.
 When used in interactive mode, the user will not be prompted for those either.
 
 
- **update disclaimer with tested linux**

Updating the initial disclaimer with the distros and versions the script has been tested on

pevma commented:
Can you please adjust the commit msg: `short: longer`.
Do we need doc update with those?

pevma commented:
When using a command like that:
```
./easy-setup.sh  --non-interactive -i tppdummy0 --iA  --es-memory 4G  && docker-compose up -d
```
the script gets into infinite loop when it comes to the interface - if that interface is not up or available/present. Might worth to put in a sanity check to be on the safe side.

pevma commented:
Also when spun up , If you open Scirius Hunt, after you click on a couple of filter sets it reverts back to the login page.

pevma commented on docker/easy-setup.sh:
```diff
@@ -41,15 +41,25 @@ function Help(){
     echo -e " --scirius-version <version>"
     echo -e "       Defines the version of scirius to use. The version can be a branch name, a github tag or a commit hash. Default is 'master'\n"
     echo -e " --elk-version <version>"
     echo -e "       Defines the version of the ELK stack to use. Default is '7.12.0'. The version should match a tag of Elasticsearch, Kibana and Logstash images on the dockerhub\n"
```
Default version currently is 7.15.1

pevma reviewed (commented):
Actually disregard the previous comment (Scirius above) - I had an existing localhost proxy interfering, sorry about the noise.

yodapotatofly commented:
> When using a command like that:
> ```
> ./easy-setup.sh  --non-interactive -i tppdummy0 --iA  --es-memory 4G  && docker-compose up -d
> ```
> the script gets into infinite loop when it comes to the interface - if that interface is not up or available/present. Might worth to put in a sanity check to be on the safe side.

fixed

> Default version currently is 7.15.1

fixed

pevma commented:
One thing to consider is that ES memory can only be done during the first time setup. If you do composer down then up again, the installation will default to 2gb. Is my understanding correct ?

pevma commented:
My bad on my previous comment - it will get written into `.env`.
